vm/state_tree/src/lib.rs

```diff
@@ -47,56 +199,52 @@ where
     }
 
     /// Get actor state from an address. Will be resolved to ID address.
-    pub fn get_actor(&self, addr: &Address) -> Result<Option<ActorState>, String> {
+    pub fn get_actor(&self, addr: &Address) -> Result<Option<ActorState>, Box<dyn StdError>> {
         let addr = match self.lookup_id(addr)? {
             Some(addr) => addr,
             None => return Ok(None),
         };
 
         // Check cache for actor state
-        if let Some(actor_state) = self.actor_cache.read().get(&addr) {
-            return Ok(Some(actor_state.clone()));
+        if let Some(actor_state) = self.snaps.get_actor(&addr) {
+            return Ok(Some(actor_state));
         }
 
         // if state doesn't exist, find using hamt
         let act = self.hamt.get(&addr.to_bytes()).map_err(|e| e.to_string())?;
 
         // Update cache if state was found
         if let Some(act_s) = &act {
-            self.actor_cache.write().insert(addr, act_s.clone());
+            self.snaps.set_actor(addr, act_s.clone())?;
         }
 
         Ok(act)
     }
 
     /// Set actor state for an address. Will set state at ID address.
-    pub fn set_actor(&mut self, addr: &Address, actor: ActorState) -> Result<(), String> {
+    pub fn set_actor(
+        &mut self,
+        addr: &Address,
+        actor: ActorState,
+    ) -> Result<(), Box<dyn StdError>> {
         let addr = self
             .lookup_id(addr)?
             .ok_or_else(|| format!("Resolution lookup failed for {}", addr))?;
 
-        // Set actor state in cache
-        if let Some(act) = self.actor_cache.write().insert(addr, actor.clone()) {
-            if act == actor {
-                // New value is same as cached, no need to set in hamt
-                return Ok(());
-            }
-        }
-
-        // Set actor state in hamt
-        self.hamt
-            .set(addr.to_bytes().into(), actor)
-            .map_err(|e| e.to_string())?;
-
-        Ok(())
+        self.snaps.set_actor(addr, actor)
     }
 
     /// Get an ID address from any Address
-    pub fn lookup_id(&self, addr: &Address) -> Result<Option<Address>, String> {
+    pub fn lookup_id(&self, addr: &Address) -> Result<Option<Address>, Box<dyn StdError>> {
         if addr.protocol() == Protocol::ID {
             return Ok(Some(*addr));
         }
 
+        match self.snaps.resolve_address(addr) {
+            None => warn!("No address cached"),
```

Comment:
why should this be a warn? address not being cached is an extremely common occurance, should probably just have this as an `if let`